sdk/identity/identity/test/httpRequests.browser.ts

```diff
@@ -40,86 +35,112 @@ export function prepareMSALResponses(): RawTestResponse[] {
-  const requests: sinon.SinonFakeXMLHttpRequest[] = [];
-  const responses: RawTestResponse[] = [];
+  async restore(): Promise<void> {
+    this.sandbox.restore();
+    AzureLogger.log = this.oldLogger;
+    setLogLevel(this.oldLogLevel);
+  }
 
   /**
    * Wraps a credential's getToken in a mocked environment, then returns the results from the request,
    * including potentially an AccessToken, an error and the list of outgoing requests in a simplified format.
    */
-  async function sendIndividualRequest<T>(
+  async sendIndividualRequest<T>(
     sendPromise: () => Promise<T | null>,
     { response }: { response: TestResponse }
   ): Promise<T | null> {
     /**
      * Both keeps track of the outgoing requests,
      * and ensures each request answers with each received response, in order.
      */
-    server.respondWith((xhr) => {
-      requests.push(xhr);
+    this.server.respondWith((xhr) => {
+      this.requests.push(xhr);
       xhr.respond(response.statusCode, response.headers, response.body);
     });
     const promise = sendPromise();
-    server.respond();
-    await clock.runAllAsync();
+    this.server.respond();
+    await this.clock.runAllAsync();
     return promise;
   }
 
   /**
    * Wraps the outgoing request in a mocked environment, then returns the error that results from the request.
    */
-  async function sendIndividualRequestAndGetError<T>(
+  async sendIndividualRequestAndGetError<T>(
     sendPromise: () => Promise<T | null>,
     response: { response: TestResponse }
   ): Promise<Error> {
-    return getError(sendIndividualRequest(sendPromise, response));
+    return getError(this.sendIndividualRequest(sendPromise, response));
   }
 
   /**
    * Wraps a credential's getToken in a mocked environment, then returns the results from the request.
    */
-  const sendCredentialRequests: SendCredentialRequests = async ({
+  async sendCredentialRequests({
     scopes,
     getTokenOptions,
     credential,
     insecureResponses = [],
     secureResponses = [],
-  }) => {
-    responses.push(...[...insecureResponses, ...secureResponses]);
-    server.respondWith((xhr) => {
-      requests.push(xhr);
-      if (!responses.length) {
+  }: {
+    scopes: string | string[];
+    getTokenOptions?: GetTokenOptions;
+    credential: TokenCredential;
+    insecureResponses?: RawTestResponse[];
+    secureResponses?: RawTestResponse[];
+  }): Promise<{
+    result: AccessToken | null;
+    error?: RestError;
+    requests: {
+      url: string;
+      body: string;
+      method: string;
+      headers: Record<string, string>;
+    }[];
+  }> {
+    this.responses.push(...[...insecureResponses, ...secureResponses]);
+    this.server.respondWith((xhr) => {
+      this.requests.push(xhr);
+      if (!this.responses.length) {
         throw new Error("No responses to send");
       }
-      const { response, error } = responses.shift()!;
+      const { response, error } = this.responses.shift()!;
```

Comment:
if this happens to be null, will the program exit abrubptly?

Reply:
Yes! I will improve this to throw if there are no responses left. Thank you!

Reply:
Actually, in this case it wouldn’t be possible since we check if there are no responses left above, but your comment helped me see that a similar check was missing on the Node side: https://github.com/Azure/azure-sdk-for-js/pull/19824/commits/a61450e9b2d566be693837a1281d64381dd5270b